corehq/apps/app_manager/views/modules.py

```diff
@@ -0,0 +1,717 @@
+    case_property_builder = _setup_case_property_builder(app)
+    fixture_columns = [
+        field.field_name
+        for fixture in FixtureDataType.by_domain(app.domain)
+        for field in fixture.fields
+    ]
+    case_type = module.case_type
+    form_options = _case_list_form_options(app, module, case_type)
+    # http://manage.dimagi.com/default.asp?178635
+    allow_with_parent_select = app.build_version >= '2.23' or not module.parent_select.active
+    allow_case_list_form = _case_list_form_not_allowed_reason(
+        module,
+        AllowWithReason(allow_with_parent_select, AllowWithReason.PARENT_SELECT_ACTIVE)
+    )
+    return {
+        'parent_modules': _get_parent_modules(app, module,
+                                             case_property_builder, case_type),
+        'fixtures': _get_fixture_types(app.domain),
+        'fixture_columns': fixture_columns,
+        'details': _get_module_details_context(app, module,
+                                               case_property_builder,
+                                               case_type),
+        'case_list_form_options': form_options,
+        'case_list_form_not_allowed_reason': allow_case_list_form,
+        'valid_parent_modules': [parent_module
+                                 for parent_module in app.modules
+                                 if
+                                 not getattr(parent_module, 'root_module_id',
+                                             None) and
+                                 not parent_module == module],
+        'child_module_enabled': toggles.BASIC_CHILD_MODULE.enabled(app.domain)
+    }
+
+
+def _get_report_module_context(app, module):
+    def _report_to_config(report):
+        return {
+            'report_id': report._id,
+            'title': report.title,
+            'charts': [chart for chart in report.charts if
+                       chart.type == 'multibar'],
+            'filter_structure': report.filters,
+        }
+
+    all_reports = ReportConfiguration.by_domain(app.domain)
+    all_report_ids = set([r._id for r in all_reports])
+    invalid_report_references = filter(
+        lambda r: r.report_id not in all_report_ids, module.report_configs)
+    warnings = []
+    if invalid_report_references:
+        module.report_configs = filter(lambda r: r.report_id in all_report_ids,
+                                       module.report_configs)
+        warnings.append(
+            gettext_lazy(
+                'Your app contains references to reports that are deleted. These will be removed on save.')
+        )
+    return {
+        'all_reports': [_report_to_config(r) for r in all_reports],
+        'current_reports': [r.to_json() for r in module.report_configs],
+        'invalid_report_references': invalid_report_references,
+        'warnings': warnings,
+    }
+
+
+def _get_fixture_types(domain):
+    # TODO: Don't hit the DB here and when getting fixture columns
```

Comment:
yes, still relevant - this gets called twice for one request in some circumstances